plugins/woocommerce/includes/admin/helper/class-wc-helper-admin.php

```diff
@@ -0,0 +1,85 @@
+<?php
+/**
+ * WooCommerce Admin Helper - React admin interface
+ *
+ * @package WooCommerce\Admin\Helper
+ */
+
+if ( ! defined( 'ABSPATH' ) ) {
+	exit;
+}
+
+/**
+ * WC_Helper Class
+ *
+ * The main entry-point for all things related to the Helper.
+ * The Helper manages the connection between the store and
+ * an account on WooCommerce.com.
+ */
+class WC_Helper_Admin {
+
+	/**
+	 * Loads the class, runs on init
+	 *
+	 * @return void
+	 */
+	public static function load() {
+		add_filter( 'woocommerce_admin_shared_settings', array( __CLASS__, 'add_marketplace_settings' ) );
+	}
+
+	/**
+	 * Pushes settings onto the WooCommerce Admin global settings object (wcSettings).
+	 *
+	 * @param mixed $settings The settings object we're amending.
+	 *
+	 * @return mixed $settings
+	 */
+	public static function add_marketplace_settings( $settings ) {
+		$settings['wccomHelper'] = array(
+			'isConnected' => WC_Helper::is_site_connected(),
+			'connectURL'  => self::get_connection_url(),
+			'userEmail'   => WC_Helper_Options::get( 'auth_user_data' ) ? WC_Helper_Options::get( 'auth_user_data' )['email'] : '',
+			'userAvatar'  => WC_Helper_Options::get( 'auth_user_data' ) ? get_avatar_url( WC_Helper_Options::get( 'auth_user_data' )['email'], 48 ) : '',
+		);
+		return $settings;
+	}
+
+	/**
+	 * Generates the URL for connecting or disconnecting the store to/from WooCommerce.com.
+	 * Approach taken from existing helper code that isn't exposed.
+	 *
+	 * @return string
+	 */
+	public static function get_connection_url() {
+		// No active connection.
+		if ( ! WC_Helper::is_site_connected() ) {
+			$connect_url = add_query_arg(
+				array(
+					'page'              => 'wc-addons',
+					'section'           => 'helper',
+					'wc-helper-connect' => 1,
+					'wc-helper-nonce'   => wp_create_nonce( 'connect' ),
+				),
+				admin_url( 'admin.php' )
+			);
+
+			// TODO: this may be needed to make the URL work.
+			// include WC_Helper::get_view_filename( 'html-oauth-start.php' );
```

Comment:
This will hopefully silence the [PHPCS warning](https://github.com/woocommerce/woocommerce/actions/runs/5715945454/job/15486367120?pr=39383) for the moment.

```suggestion
			// phpcs:ignore Generic.Commenting.Todo.TaskFound
			// TODO: this may be needed to make the URL work.
			// include WC_Helper::get_view_filename( 'html-oauth-start.php' ).
```

Reply:
Happy to do this, but do we want to silence this? This should be resolved fully before we finalise the project and so the warning is a reminder. It could be a little easy to miss the comment by the time we merge. 🤔

Reply:
Ya, that was at the back of my mind when I suggested this. We could merge the PR without it.